frb_example/pure_dart/rust/src/api.rs

```diff
@@ -579,12 +579,9 @@ pub fn register_event_listener(listener: StreamSink<Event>) -> Result<()> {
 }
 
 pub fn close_event_listener() {
-    if let Some(ref listener) = *EVENT_LISTENER.lock().unwrap() {
-        listener.close();
-    } else {
-        return;
+    if let Ok(Some(listener)) = EVENT_LISTENER.lock().map(|guard| guard.take()) {
+        listener.close()
     }
-    (*EVENT_LISTENER.lock().unwrap()) = None;
 }
```

Comment:
May I know why the code changes - the old one seems not to have problems

Reply:
I felt that it would be more concise this way, but I can revert it if it's not too important.

Reply:
I see. Neat solution!